### ISSUE TITLE
Design handouts template

### ISSUE THREAD

**Issue by cassiemc** [mozillascience/working-open-workshop#11]
- [x] make a rough wireframe
- [x] figure out where this will live (github? gitbook? pdf? ~~thimble?~~)
- [x] design styles

Creative brief is here: https://docs.google.com/document/d/1EAYGXIcbawbLMWedDhKWH2-zPovxSMTx_YOkNpjDabI/edit#

Sample content is here: https://public.etherpad-mozilla.org/p/WOW-personas-pathways-handout

**Comment by cassiemc** [comment]
@zee-moz Can you list here the range of things the handout should accommodate? sections, etc?

**Comment by zee-moz** [comment]
https://github.com/mozillascience/learning/blob/master/resource-template.md >>this is the standard template. 

I will provide more as we make them...

**Comment by cassiemc** [comment]
@zee-moz I still can't seem to assign @flukeout or update milestones, maybe try adding me as an admin? Or you can assign this and #12 to Luke :)

**Comment by flukeout** [comment]
**Some handy links**
- [Mozilla Science Lab Site](https://www.mozillascience.org/)
- [Personas Handout Content](https://public.etherpad-mozilla.org/p/WOW-personas-pathways-handout)
- [Handout Content Template](https://github.com/mozillascience/learning/blob/master/resource-template.md)
- [Working Open Gitbook](http://mozillascience.github.io/open-science-leadership-workshop/index.html)
- [Curriculum Template Thimble Project](https://d157rqmxrxj6ey.cloudfront.net/flukeout/24615/#overview)
- [Science Fox Artwork](https://drive.google.com/open?id=0B_CPRh4f3yyqdkxDeVlDNDFlNlU)

**Comment by flukeout** [comment]
@acabunoc Can you share any thoughts you've had about what a good way to host this content would be? We were thinking of using github pages.

**Comment by abbycabs** [comment]
Github pages are good depending on how easy it would be to add / update / remix content. Markdown would be nice!

**Comment by abbycabs** [comment]
I initially suggested gitbook for ease of setup & content editing since I didn't think we'd have time to build something else. I do like gh pages, though.

**Comment by flukeout** [comment]
@acabunoc The sense I got from @zee-moz is that as a team, you're pushing participants to get familiar with git, so adding and updating content would probably take the form of forks & pull requests (etc.) - is that too high bar in your eyes?

Take a looks at https://d157rqmxrxj6ey.cloudfront.net/flukeout/24615/#overview - it's a Thimble template we're working on for our curriculum - I was thinking of creating something similar for these handouts, but it would be hosted on github instead.

Using markdown would limit the aesthetic quality of these materials, but would make them more approachable in terms of editing.

**Comment by abbycabs** [comment]
forks & pulls are fine! I'm more concerned about them editing HTML (which we do not cover, and most of our team creating the content is not comfortable with) -- hence the markdown ask

**Comment by flukeout** [comment]
@acabunoc Are there methods or ways you know of that take markdown and parse it to allow us to use CSS to style it? I found [http://strapdownjs.com/](http://strapdownjs.com/) - which appears to do something like that. This would allow people to make edits in markdown, but give us more flexibility visually.

Just a thought. It would also affect our layout options for the handouts.

**Comment by abbycabs** [comment]
@flukeout 
http://strapdownjs.com/ looks good! The only other thing I can think of is Jekyll which plays nice with gh-pages, but I think either would work :smile: 

Thanks so much for working with our constraints!

**Comment by flukeout** [comment]
Here's a pass at potential layout using the Personas content.

![image](https://cloud.githubusercontent.com/assets/25212/12466240/e5231ffe-bf87-11e5-98e8-f9a4b8f10224.png)

**Thoughts & Questions**
- We could make the side-nav persistent and scroll the page the the right section when it's clicked
  - And also update it to the right section when a user scrolls through
- Alternatively, we could divide this page so that only one step of the activity is visible at a time, like with the curriculum template. 
  - The steps seem quite short in this handout though, so maybe it's not worth it.
- How many steps, and how elaborate will they be, do you anticipate having per curriculum?
- If the Steps aren't as important to the structure of the document, we could put some of the meta data (like the audience, time required, materials, etc.) in the sidebar.
- Do we want to build in a "update this" or "fork this content" CTA on this document somewhere prominently?
- Are there preset difficulty levels in this content (like beginner, intermediate, advanced?) If so, maybe we can do a little graphical treatment that indicates that there are three levels, like a little slider or something.
- Im using the Fox as a stand-in for a potential placement for imagery related to this piece - we could use an icon or illustration in the style of the new icons we're using.

Curious to get some first impressions @zee-moz @cassiemc and @acabunoc - any realizations or thoughts in reaction to this first draft? Things we're forgetting or not considering, etc?

I'll explore some other directions that take into account some more of the styles and elements from the mozilla science site. Another option is to look at the work that @kristinashu is doing in the new fellows work here - https://redpen.io/p/wt1d64bfb93fa6a6b8

**Comment by zee-moz** [comment]
@flukeout this looks great!

i like the idea of the persistent side nav and scrolling to steps for these short handouts. these activities are short-- designed to do in under 30 minutes. we will also have longer lessons with more steps-- but i think the persistent nav can work for that, too-- i like the idea of being able to see where you are in the activity sequence, and be able to back up if need be.  

i like having the meta stuff up top in the main page, when it's highlighted there it really frames the experience. 

i edited the copy of the handout a bit yesterday, and now it has example in it-- is there a way to highlight examples somehow in the body of the page? i may move this example to be on the same page as the step it's related to, or add a second one there... seeing this structure is a good reminder that it's probably or might be helpful to keep the intro light and provide explication under the steps. 

Update and fork-- I think this sounds good. we'll need to create some guidelines for doing so but @acabunoc and @auremoser and i can figure that out. 

yes, three pre-set levels! right now most of our stuff is novice, but we'll have more as we expand. 

additional thoughts: having those three columns for format, audience, and materials is helpful-- and i think that structure will keep those sections, and the goal section, from getting too wordy. 

love that fox!!!

i will send on more content for you to play with later on today!  thanks again, this looks so great!

@stephwright @arlissc @kaythaney comments?

**Comment by kaythaney** [comment]
this looks great - thanks @flukeout + @zee-moz! 

as for the side nav - loving the walk through on steps. this might be a bit feature creep-y for the short term, but might be worth thinking of how in that side bar we could surface other exercises people should jump to afterwards (or that may be part of a set). can see that functionality as useful down the road to help with discoverability of other lessons. :tada: :+1: :clap: thank you, though!

**Comment by stephwright** [comment]
Dang @flukeout!  This looks great!
I concur w/ @zee-moz re: side nav being persistent w/ scrolling (speaking as one who gets lost w/i websites sometimes).  Ditto on the meta stuff.

Thinking of what I'm planning for WOW, is there a way / place to link to add'l materials?  I'm thinking of a template for my exercise to help them create a data reuse plan.   I can probably make it work without but just thinking it would help them get up and running more quickly on a complicated subject if they have a template they can printout and use.

So great to see the fox in there!  Rock star! :star2:

**Comment by auremoser** [comment]
I like the update and fork if we want to get the community involved with contributions, would be a good model to re-structure study group lessons too, if we have a lovely template like this we could swap the css and keep the same look/feel across MSL programs, because everyone has training. 

We had a similar site/page for tutorials for the  [CartoDB Academy](http://academy.cartodb.com/), so thinking forward about what a little library of lessons might look like, it might be cool to check that out. I'm happy to help work out some of the technical workflows (academy was basically a jekyll blog), we didn't let people fork/update but we did have a "community lesson" section. 

Thank you so much @flukeout , you rock!

**Comment by flukeout** [comment]
**Thanks for the feedback**
- @zee-moz I'll update the design to include more of the content types from the outline (like the Example)  - I wanted to start with just a rough direction
- @kaythaney - We can plan around a static navigation for now, and evaluate how much work it would be to make it persistant (I don't think very much, though)
- @stephwright - Yeah, I didn't include the Resources section in the navigation, I'll mock up a more extensive version that includes all of the content so far so you can take a look.

@cassiemc has scheduled a meeting for all of us to talk about the technical details, there are some options (HTML vs Markdown, for example) that affect the design options we have, so I'm looking forward to getting that settled.

**Comment by flukeout** [comment]
Some options for an **Example** callout...

![image](https://cloud.githubusercontent.com/assets/25212/12492793/d4f1cc70-c036-11e5-870e-6469aa1fdd8b.png)

**Comment by zee-moz** [comment]
These are good! I think I prefer the top and bottom ones. I think the highlighted text box great... The second option, with the arrow-bar creates a visual relationship with what's happening in the nav at left, and that's maybe confusing? I think we don't really need the bars, just the highlight is great.

**Comment by stephwright** [comment]
I like the first & second myself but that's cuz my old eyes prefer the higher contrast of the dark background with lighter text.  I defer to @zee-moz as she has more design chops than I do. Looks nice!

**Comment by flukeout** [comment]
Okay cool - I'm working on the Agenda Steps portion now - when it's ready I'll move it to a redpen for better critique, discussion.

**Comment by flukeout** [comment]
Dropping in some styles for the **Agenda Section** ahead of our call...

![image](https://cloud.githubusercontent.com/assets/25212/12495691/057f9584-c046-11e5-871b-285fa8df29c6.png)

**Comment by zee-moz** [comment]
@flukeout  here's a slightly different kind of resource-- a tip sheet. Can it fit in this template? 

https://public.etherpad-mozilla.org/p/sciencelab-facilitation-handout

It's actually broken down into sections (before the event, after the event, etc) so that may lend itself to the sidebar form. Materials aren't really applicable for this resource, and the format might not be applicable either... 

Thanks for all your great work! Z

**Comment by zee-moz** [comment]
@auremoser btw, I love that library of resources-- that's exactly what I've been envisioning for MSL...

**Comment by abbycabs** [comment]
An example of markdown behind a lesson:

Markdown: https://github.com/swcarpentry/sql-novice-survey/blob/gh-pages/01-select.md
Rendered: http://swcarpentry.github.io/sql-novice-survey/01-select.html

**Comment by flukeout** [comment]
Here are the designs above in redpen...
https://redpen.io/wnec8079b1ced99e8a

**Things to consider**
- How do we link back to other resources?
- Will the multi-column layout work for the glossary and resources but still be easy to edit?
  - I'll experiment with CSS columns
  - It's important for the glossary to stay alphabetical

**Comment by flukeout** [comment]
Here's a first crack at a working version with a persistent navigation and the content from the etherpad, which I think still isn't finished...

http://flukeout.github.io/science-template/

I created [my own repo for now](https://github.com/flukeout/science-template).

cc @zee-moz @acabunoc It's still rough around the edges, so let me know your thoughts at this point. 

We're still gathering feedback from the design team on the redpen.

**Comment by flukeout** [comment]
**To-do**
- Review and incorporate redpen feedback
  - [x] Investigate smooth scroll when side-navigation is used
  - [x] Make styling adjustments
  - [x] Try out a couple of alternate colors 
- [x] Fix column widths (especially for the description section)
- [x] Use the mozilla science favicon
- [x] Clean up old CSS
- [ ] Create Print styles CSS
- [x] Investigate the column-wrap thing for Firefox (to make sure that a definition won't wrap to column 2 when the term is still in column 1) - works fine in Chrome
- [x] See if I can reduce which font-weights we load from Google for Fira
- [x] ...
- [x] Profit

**Comment by cassiemc** [comment]
@flukeout I left some comments in redpen. Main thing that would be good to see is another colour variation and if there is time, some finesse in the typography and spacing.

**Comment by flukeout** [comment]
@cassiemc Thanks for the feedback! Will try a couple of alternate "main" colors within the current design and address the feedback in the working "live" version.

**Comment by cassiemc** [comment]
Sounds good!

**Comment by zee-moz** [comment]
@flukeout I revised and added to the User Personas sheet-- per some discussions with a real-world user of this content I added some more explanation and an additional step (list your solutions). I also defined a term and provided links to relevant content, although those items don't yet exist (but will soon!)
https://public.etherpad-mozilla.org/p/WOW-personas-pathways-handout
let me know if you have any questions! thanks, z

**Comment by flukeout** [comment]
@zee-moz Okay, great - I've update the doc with the latest content http://flukeout.github.io/science-template/

**Comment by flukeout** [comment]
Quick - side by side of an alternative color scheme, will try a couple more tomorrow morning and we'll settle one by EOD...

![image](https://cloud.githubusercontent.com/assets/25212/12600449/b838eb74-c44e-11e5-9ccc-1fcd243f2487.png)

**Comment by cassiemc** [comment]
Nice @flukeout. The blue is feeling more on-brand from my perspective. Maybe try the light blue from the lab coat for the persona background? Green makes it stand out probably more than it needs to.

**Comment by flukeout** [comment]
**Latest version** - http://flukeout.github.io/science-template/

This is pretty much good to go.

@acabunoc let's sync up and decide how to host & port this to markdown.

**Comment by zee-moz** [comment]
hey, sorry for this delay-- one small item of feedback from our team-- the MSL logo seems a bit small and hanging off too far on the right. Can we make it a bit bigger and feeling less lonely out there?
Thanks!

**Comment by cassiemc** [comment]
@flukeout did you try aligning the logo to the left, as per @sabrinang's redpen suggestion? I dig the color and spacing updates, thanks.

**Comment by stephwright** [comment]
The Glossary part looks like it's confined into a column on the left.  Is there a design reason for that?  Just thinking of the exercise I just did with four words in the Glossary and that would take up quite a bit of real estate if confined to the left side.

**Comment by zee-moz** [comment]
@stephwright I could be wrong, but I think that's because there's only one definition in this particular handout. If there were more, it would take up two columns (at least it did in the previous design). But it looked weird spreading a single definition across both columns, so that's not happening in this case. @flukeout can you confirm?

**Comment by flukeout** [comment]
@stephwright @zee-moz - Yes, the glossary spans two columns, so with one definition it doesn't look as nice. We could address it by detecting how many definitions there are via javascript and adjusting the layout, but that seems like overkill at this point.

@cassiemc No - haven't tried the Logo changes yet, will do a trial.

**Comment by stephwright** [comment]
Just gave the new template a whirl with my lesson here: http://mozillascience.github.io/working-open-workshop/data_reuse/

Looks like the Glossary didn't feel like cooperating on the wrapping aspect.  I also attempted a nested unordered list in the Follow-Up Resources section and it wouldn't let me nest so I just left as it was.  Not a big deal.

Thanks so much for doing this @flukeout !  The lesson looks SO much prettier than it did in markdown.

**Comment by cassiemc** [comment]
So close to closing! @flukeout do you want to do that and open up new bugs for further issues?

**Comment by cassiemc** [comment]
btw @stephwright your lesson looks great!

**Comment by stephwright** [comment]
Aww... thanks @cassiemc :)  You just made my day

**Comment by stephwright** [comment]
:cookie: for you

**Comment by flukeout** [comment]
@acabunoc @zee-moz @stephwright I'm going to close this issue now, since the first handout is using the template. Let's file follow-up issues and feature requests in separate issues, ya?
